File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {

        if(x == 1) return 1;
        int l = 1, r = x;
        while(l < r) {
            int m = (r - l) / 2 + l;
            if(m > x / m) r = m;
            else l = m + 1;
        }
        return l - 1;
    }
}

File: 0069-sqrtx/0069-sqrtx.js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    if (x === 1) return 1;
    let l = 1, r = x;
    while (l < r) {
        let m = Math.floor((r - l) / 2) + l;
        if (m > Math.floor(x / m)) r = m;
        else l = m + 1;
    }
    return l - 1;
};